### PR TITLE
Re-implement Corelight table structure

### PR DESCRIPTION
**Pull request by ekoyle**
Required items, please complete
   
   Change(s):
   - Change the backend data format for the Corelight solution
   
   Reason for Change(s):
   - Low performance from existing solution

   Version Updated:
   - Yes

   Testing Completed:
   - Yes

   Checked that the validations are passing and have addressed any issues that are present:
   - Need Help
   
I'm not sure what the validations are complaining about because there is too much output on my workstation for me to see the errors.

### CODE REVIEW

**Review by oshezaf — CHANGES_REQUESTED** [review]
- The parsers starting with ASIM, while sharing some naming conversion with ASIM, are not ASIM compliant and the ASIM prefix should be removed from the names.
- The other parsers have generic names such as "dhcp" and "conn" which may overlap with functions users already have in their workspaces leading to hard to nail serious issues with queries. Those parses need to be prefix with a standatd, corelight related, prefix.

**Comment by ekoyle** [comment]
@microsoft-github-policy-service agree company="Corelight"

**Comment by ekoyle** [comment]
I have dropped the ASIM-like parsers for now and added a `corelight_` prefix to the rest.

I'm not sure why the KqlValidations test is failing.  It looks like it doesn't like the reference to any of the `corelight_*` parsers which are defined as part of this solution.

**Comment by v-sabiraj** [comment]
@devikamehra, can you please review Analytic rules, thanks.

**Review by devikamehra — CHANGES_REQUESTED** [review]

File: Solutions/Corelight/Analytic Rules/CorelightC2RepetitiveFailures.yaml [hunk 1]
```diff
@@ -16,16 +17,15 @@ tactics:
   - CommandAndControl
 query: |
   let threshold = 25;
-  Corelight
-  | where EventType =~ 'dns'
-  | where DnsResponseCodeName in~ ('NXDOMAIN', 'SERVFAIL')
-  | summarize count() by DnsQueryName, SrcIpAddr, bin(TimeGenerated, 5m)
+  union corelight_dns, corelight_dns_red
+  | where rcode_name in~ ('NXDOMAIN', 'SERVFAIL')
```
[the remark this answers is not in the document]
Comment (ekoyle) [reply Solutions/Corelight/Analytic Rules/CorelightC2RepetitiveFailures.yaml]: I changed it to use `rcode` instead

**Comment by v-sabiraj** [comment]
@ekoyle, can you please fix the validations, thanks.

**Comment by ekoyle** [comment]
@v-sabiraj I'm not sure how to fix the KqlValidations.  It looks like it is not finding the new parsers for some reason, although the queries run fine with the solution installed.  I'm wondering if I haven't specified dependencies properly?

This is what I have in the yaml:

```
requiredDataConnectors:
  - connectorId: Corelight
    dataTypes:
      - corelight_conn
      - corelight_conn_red
```

I am seeing a lot of errors like the following:

```
2023-03-08T18:43:08.7801362Z [xUnit.net 00:00:07.69]     Kqlvalidations.Tests.KqlValidationTests.Validate_DetectionQueries_HaveValidKql(fileName: "CorelightForcedExternalOutboundSMB.yaml", encodedFilePath: "L2hvbWUvdnN0cy93b3JrLzEvcy9Tb2x1dGlvbnMvQ29yZWxpZ2"...) [FAIL]
2023-03-08T18:43:09.8360644Z   Failed Kqlvalidations.Tests.KqlValidationTests.Validate_DetectionQueries_HaveValidKql(fileName: "CorelightForcedExternalOutboundSMB.yaml", encodedFilePath: "L2hvbWUvdnN0cy93b3JrLzEvcy9Tb2x1dGlvbnMvQ29yZWxpZ2"...) [182 ms]
2023-03-08T18:43:09.8361499Z   Error Message:
2023-03-08T18:43:09.8362342Z    Template Id: 73f23aa2-5cc4-4507-940b-75c9092e9e01 is not valid in Line: 1 col: 7
2023-03-08T18:43:09.8363342Z                     Errors: The name 'corelight_conn' does not refer to any known table, tabular variable or function., Code: 'KS204', Severity: 'Error', Location: '6..20',The name 'corelight_conn_red' does not refer to any known table, tabular variable or function., Code: 'KS204', Severity: 'Error', Location: '22..40'
2023-03-08T18:43:09.8363949Z Expected: True
2023-03-08T18:43:09.8364173Z Actual:   False
2023-03-08T18:43:09.8364355Z   Stack Trace:
2023-03-08T18:43:09.8374402Z      at Kqlvalidations.Tests.KqlValidationTests.ValidateKql(String id, String queryStr, Boolean ignoreNoTabularExpressionError) in /home/vsts/work/1/s/.script/tests/KqlvalidationsTests/KqlValidationTests.cs:line 211
2023-03-08T18:43:09.8375519Z    at Kqlvalidations.Tests.KqlValidationTests.Validate_DetectionQueries_HaveValidKql(String fileName, String encodedFilePath) in /home/vsts/work/1/s/.script/tests/KqlvalidationsTests/KqlValidationTests.cs:line 63
2023-03-08T18:43:41.0996138Z [xUnit.net 00:00:40.01]     
```

**Comment by v-sabiraj** [comment]
let me check on that.

**Comment by ekoyle** [comment]
I got past the KqlvalidationsTests, but now it is failing on DataConnectorValidations.  Is there any documentation on what is needed for that test?

**Comment by ekoyle** [comment]
I found a workaround, for the last test.

**Comment by v-sabiraj** [comment]
@devikamehra, can you please check the changes done, thanks.

**Comment by v-prasadboke** [comment]
Hello @devikamehra any updates on the following

**Comment by v-rbajaj** [comment]
Hi @devikamehra, can you please share some update?

**Comment by v-sabiraj** [comment]
@ekoyle, can you please add sample to test parser and data connector, thanks.

**Comment by ekoyle** [comment]
I currently have a single file with generated samples and a script to send them to the separate tables.  Would you prefer ~100 sample files under `Sample Data` or the single file with a python script in our solution directory?

**Comment by devikamehra** [comment]
@v-sabiraj Can you please reply to ekoyle

**Comment by ekoyle** [comment]
I just added `Sample Data/Corelight/*.json` for sample data.  Let me know if you need something different.

**Comment by v-prasadboke** [comment]
Hello @v-sabiraj please provide your feedback on this

**Comment by v-prasadboke** [comment]
Hello @ekoyle please remove failed validations

**Comment by ekoyle** [comment]
@v-prasadboke validations are passing now

**Comment by v-atulyadav** [comment]
Hi @v-sabiraj, please reply to ekoyle.

**Comment by v-prasadboke** [comment]
Hello @v-sabiraj  please provide your reply to the author

**Comment by v-prasadboke** [comment]
Hello @v-sabiraj waiting for your feedback

**Comment by v-prasadboke** [comment]
Looking into this

**Comment by v-prasadboke** [comment]
Hello @ekoyle sorry for the late reply, while deploying the solution via custom deployment, Im facing an issue due to template variables.
The limit set for tempalte variable in maintemplate is 512 but in this solution its 982.
But i think due to the number of parser this solution consist is causing the error.

Can you please describe the reason behind adding 100+ parsers in the solution

**Comment by ekoyle** [comment]
We are trying to work around the way the backend modifies our schema for custom logs. We have one parser per table (of which we have about 100) that is renaming the columns back to what we sent over (removing _s/_d/etc from the end of column names).  We are also planning to use this to keep our user-facing data format the same even if the backend table structure changes (if we move to the new DCR-based API, for example), as we expect customers to build their own dashboard, etc.
We initially tried putting everything into one table, but there are too many columns in our data (512 column limit).
We also looked at using the new DCR-based API, but customers can change/extend their log format, and that would cause data loss for any columns that aren't handled by our solution as the new data collection API doesn't seem to have a way to handle dynamic schema changes.
Could I get more information on where you are seeing so many template variables?

File: Solutions/Corelight/Package/mainTemplate.json [hunk 2]
```diff
@@ -28,2891 +28,3952 @@
...
     "workbook1-name": {
       "type": "string",
       "defaultValue": "Corelight",
       "minLength": 1,
       "metadata": {
         "description": "Name for the workbook"
       }
-    }
-  },
-  "variables": {
```
Comment (v-prasadboke) [inline Solutions/Corelight/Package/mainTemplate.json:40]: Instances set under this parameter ( variables )  are template variabels which are crossing the limit of 512 
MicrosoftTeams-image (20)

**Comment by ekoyle** [comment]
I only see about 270 variables in the mainTemplate.json I submitted.

**Comment by v-prasadboke** [comment]
Hello @ekoyle you havent repackaged the solution properly.
Please go through this readme file for repackaging of the solution.
https://github.com/Azure/Azure-Sentinel/blob/master/Tools/Create-Azure-Sentinel-Solution/README.md

**Comment by ekoyle** [comment]
I used `Tools/Create-Azure-Sentinel-Solution/createSolution.ps1` to package the solution.  It did require some changes because the versions were too old, as well as some changes to allow more than one parser to be created (I added the function alias to the data parser name).  Should I include those in this PR?

```
diff --git a/Tools/Create-Azure-Sentinel-Solution/createSolution.ps1 b/Tools/Create-Azure-Sentinel-Solution/createSolution.ps1
index d1c7d223c..4c67d22aa 100644
--- a/Tools/Create-Azure-Sentinel-Solution/createSolution.ps1
+++ b/Tools/Create-Azure-Sentinel-Solution/createSolution.ps1
@@ -245,7 +245,7 @@ foreach ($inputFile in $(Get-ChildItem $path)) {
                             name       = "[parameters('workbook$workbookCounter-id')]";
                             location   = "[parameters('workspace-location')]";
                             kind       = "shared";
-                            apiVersion = "2021-08-01";
+                            apiVersion = "2022-04-01";
                             properties = [PSCustomObject] @{
                                 displayName    = "[concat(parameters('workbook$workbookCounter-name'), ' - ', parameters('formattedTimeNow'))]";
                                 serializedData = $serializedData;
@@ -772,7 +772,7 @@ foreach ($inputFile in $(Get-ChildItem $path)) {
 
                                 $savedSearchResource = [PSCustomObject]@{
                                     type = "Microsoft.OperationalInsights/workspaces/savedSearches";
-                                    apiVersion = "2020-08-01";
+                                    apiVersion = "2022-10-01";
                                     name = "[concat(parameters('workspace'),'/',parameters('$savedSearchIdParameterName'))]";
                                     properties = [PSCustomObject]@{
                                         category = $search.properties.category;
@@ -894,7 +894,7 @@ foreach ($inputFile in $(Get-ChildItem $path)) {
                                 if (!$(queryResourceExists)) {
                                     $baseHuntingQueryResource = [PSCustomObject] @{
                                         type       = "Microsoft.OperationalInsights/workspaces";
-                                        apiVersion = "2020-08-01";
+                                        apiVersion = "2022-10-01";
                                         name       = "[parameters('workspace')]";
                                         location   = "[parameters('workspace-location')]";
                                         resources  = @()
@@ -927,7 +927,7 @@ foreach ($inputFile in $(Get-ChildItem $path)) {
                             }
                             $huntingQueryObj = [PSCustomObject] @{
                                 type       = "savedSearches";
-                                apiVersion = "2020-08-01";
+                                apiVersion = "2022-10-01";
                                 name       = "$solutionName Hunting Query $huntingQueryCounter";
                                 dependsOn  = @(
                                     "[variables('workspace-dependency')]"
@@ -1094,7 +1094,7 @@ foreach ($inputFile in $(Get-ChildItem $path)) {
                             $newAnalyticRule = [PSCustomObject]@{
                                 type       = "Microsoft.OperationalInsights/workspaces/providers/alertRules";
                                 name       = "[concat(parameters('workspace'),'/Microsoft.SecurityInsights/',parameters('analytic$analyticRuleCounter-id'))]";
-                                apiVersion = "2021-03-01-preview";
+                                apiVersion = "2023-02-01";
                                 kind       = "Scheduled";
                                 location   = "[parameters('workspace-location')]";
                                 properties = $alertRule;
@@ -1154,7 +1154,7 @@ foreach ($inputFile in $(Get-ChildItem $path)) {
                         if ($parserCounter -eq 1 -and !$(queryResourceExists)) {
                             $baseParserResource = [PSCustomObject] @{
                                 type       = "Microsoft.OperationalInsights/workspaces";
-                                apiVersion = "2020-08-01";
+                                apiVersion = "2022-10-01";
                                 name       = "[parameters('workspace')]";
                                 location   = "[parameters('workspace-location')]";
                                 resources  = @(
@@ -1165,14 +1165,14 @@ foreach ($inputFile in $(Get-ChildItem $path)) {
                         }
                         $parserObj = [PSCustomObject] @{
                             type       = "savedSearches";
-                            apiVersion = "2020-08-01";
-                            name       = "$solutionName Data Parser";
+                            apiVersion = "2022-10-01";
+                            name       = "$solutionName $functionAlias Data Parser";
                             dependsOn  = @(
                                 "[variables('workspace-dependency')]"
                             );
                             properties = [PSCustomObject] @{
                                 eTag          = "*";
-                                displayName   = "$solutionName Data Parser";
+                                displayName   = "$solutionName $functionAlias Data Parser";
                                 category      = "Samples";
                                 functionAlias = "$functionAlias";
                                 query         = $content;
@@ -1228,7 +1228,7 @@ foreach ($inputFile in $(Get-ChildItem $path)) {
 
                 $newMetadata = [PSCustomObject]@{
                     type       = "Microsoft.OperationalInsights/workspaces/providers/metadata";
-                    apiVersion = "2021-03-01-preview";
+                    apiVersion = "2023-02-01";
                     properties = [PSCustomObject] @{
                         version = $contentToImport.Version;
                         kind    = "Solution";
@@ -1409,4 +1409,4 @@ foreach ($inputFile in $(Get-ChildItem $path)) {
         Invoke-Expression "$armTtkFolder/download-arm-ttk.ps1"
     }
     Invoke-Expression "$armTtkFolder/run-arm-ttk-in-automation.ps1 '$solutionName'"
-}
\ No newline at end of file
+}

```

**Comment by v-prasadboke** [comment]
Hello @ekoyle, Tool we use for repackaging of the solution is " Tools/Create-Azure-Sentinel-Solution/V2/createSolutionV2.ps1 ".
Which is Version2 and you used Version1 which reflected less variables.
Please use Version2 for repackaging.

**Comment by ekoyle** [comment]
I followed the directions you gave me.  Those instructions said to use the v1 script.  You have already determined that the v2 script will not work because it is creating way too many variables for each function (like 8 variables?).
How do you expect me to resolve this? Is this is a bug in the v2 script or do you not support more than around 60 functions in a solution?

**Comment by v-prasadboke** [comment]
Hello @ekoyle, Sorry for the wrong readme file I forwarded you first. Let me check, what we can do from our side with this.

**Comment by anki-narravula** [comment]
Hi @ekoyle , I was going through this PR and just curious to understand few things and see if there are any possibilities to reduce the number of tables and parsers. Looks like these many tables / parsers create lot of clutter on customer's workspace. Can you please ping me or send me meeting invite on anknar@microsoft.com Looking forward to meet you.

**Comment by v-prasadboke** [comment]
Hello @ekoyle reduced the variable count manually, will be investigating the PR now. You can expect an update till 21 Jul 2023

File: Solutions/Corelight/Analytic Rules/CorelightC2RepetitiveFailures.yaml [hunk 3]
```diff
@@ -16,16 +17,15 @@ tactics:
   - CommandAndControl
```
Comment (v-prasadboke) [inline Solutions/Corelight/Analytic Rules/CorelightC2RepetitiveFailures.yaml:17]: Hello @ekoyle relevant techniques is missing from this detection, Please add it below tactics:

File: Solutions/Corelight/Analytic Rules/CorelightExternalProxyDetected.yaml [hunk 4]
```diff
@@ -18,15 +18,13 @@ tactics:
 relevantTechniques:
   - T1090
```
Comment (v-prasadboke) [inline Solutions/Corelight/Analytic Rules/CorelightExternalProxyDetected.yaml:19]: Missing relevant techniques here as well, Please add it below tactics:

**Comment by v-prasadboke** [comment]
And also can you please share some working images of workbook

**Comment by v-prasadboke** [comment]
Hi there, @ekoyle Could you please share some updates on the aforementioned changes requested?

**Comment by ekoyle** [comment]
The analytic rules were from our previous integration, I only updated the queries to make them work with the new table format.
What is your preferred image format for screenshots?

**Comment by v-prasadboke** [comment]
Hello @ekoyle, Can you add the latest working images of workbook in this folder **Solutions/Corelight/Workbooks/Images/Preview** by deleting the older images.
Format should be png. 

Thanks!

**Comment by devikamehra** [comment]
@rahul0216 For visibility

**Comment by v-prasadboke** [comment]
Hello @ekoyle, We are still waiting for the workbook images requested. Can you please share the same.

**Comment by v-prasadboke** [comment]
Hello @ekoyle, we are still waiting for the images can you please add them.

**Comment by ekoyle** [comment]
I have updated the screenshots.

**Comment by v-prasadboke** [comment]
Hello @ekoyle, Thank you for sharing the screenshots. I'll Examine this images and Update you about the same.

**Comment by v-prasadboke** [comment]
Hello @ekoyle, Please go to this file ' https://github.com/Azure/Azure-Sentinel/blob/master/.script/utils/dataConnector.ts '  and at the end please add "Corelight" = "Corelight".

And in this file ' https://github.com/Azure/Azure-Sentinel/blob/master/.script/tests/detectionTemplateSchemaValidation/ValidConnectorIds.json ' please add "Corelight" at the end

**Comment by v-atulyadav** [comment]
Hi @ekoyle, hope you are doing well. Just wanted to check if you got a chance to look at the suggestions shared. Please feel free to reach out to us for any queries and/or support. Thanks

**Comment by v-prasadboke** [comment]
Hello @ekoyle, Any updates on the above forementioned changes.

**Comment by ekoyle** [comment]
@v-prasadboke I have made the requested changes in the `.scripts` directory

**Comment by v-prasadboke** [comment]
Hello @ekoyle, Thank you for providing the updates and committing the changes. This will be investigated. You can expect and update till 21 Aug 2023.

**Comment by v-prasadboke** [comment]
Hello @ekoyle please a column 'id_orig_h' in this custom table [ https://github.com/Azure/Azure-Sentinel/blob/master/.script/tests/KqlvalidationsTests/CustomTables/Corelight.json ]

**Comment by v-prasadboke** [comment]
Hello @ekoyle, I've modified the maintemplate. I've changes the sentinel branding to Microsoft Sentinel. Thanks!

**Comment by v-prasadboke** [comment]
Hello @ekoyle, Any updates on the above.

**Comment by ekoyle** [comment]
These files are for the old schema and don't have much similarity to the new schema.  I'm not sure what to do with them.

https://github.com/Azure/Azure-Sentinel/blob/master/.script/tests/KqlvalidationsTests/CustomTables/Corelight.json 

Also, Corelight_CL.json

**Comment by v-prasadboke** [comment]
Hello @ekoyle, I tried adding the column to the custom table. But its still failing for that. 
We are still working on this and update you about the same before 23 August, 2023.

**Comment by v-prasadboke** [comment]
Hello @ekoyle, Sorry for the inconvenience. We are still working on this and update you about this before 28 August, 2023.

File: Solutions/Corelight/Data Connectors/Connector_Exporter_Corelight.json [hunk 5]
```diff
@@ -0,0 +1,550 @@
...
+  ],
+  "sampleQueries": [
+    {
+      "description": "Top 10 Clients (Source IP)",
+      "query": "Corelight\n\t| summarize count() by id_orig_h\n\t| top 10 by count_"
+    }
+  ],
+  "dataTypes": [
+    {
+      "name": "Corelight_v2_UNCATEGORIZED_CL",
```
Comment (v-prasadboke) [inline Solutions/Corelight/Data Connectors/Connector_Exporter_Corelight.json:22]: Hello @ekoyle, You have mentioned a parser Corelight_v2_UNCATEGORIZED_CL in the connector.
But is not located in the solution. Try adding the parser in the solution.

**Comment by v-prasadboke** [comment]
Hello @ekoyle, I've requested some changes.

**Comment by v-prasadboke** [comment]
Hello @ekoyle, Any updates on the above requested changes.

**Comment by ekoyle** [comment]
That table is to avoid losing customer-added data and new tables that
haven't made it into our solution yet.  As such, we cannot know what the
structure will be.

-- 
Eldon

**Comment by v-prasadboke** [comment]
Ok @ekoyle,
But a validation check is failing for that.

**Comment by ekoyle** [comment]
Would it be better to remove the reference entirely, even though the connector will write to that table?

**Comment by v-prasadboke** [comment]
Hello @ekoyle, If it is not a parser than should we consider it as a table. If it is a table, We recommend you to add custom table in [ .script/tests/KqlvalidationsTests/CustomTables ] this location.

If we try to remove the reference of ( Corelight_v2_UNCATEGORIZED_CL ) from the data connector we would need to remove the query as well. As that would lead to a Failed Validation check.

**Comment by v-prasadboke** [comment]
Hello @ekoyle, Can you please provide some updates on the above.
Thank You.

File: Solutions/Corelight/Data Connectors/Connector_Exporter_Corelight.json (hunk 5 above)
Comment (ekoyle) [reply Solutions/Corelight/Data Connectors/Connector_Exporter_Corelight.json]: I have removed this reference as it is not referenced anywhere else in our integration.  Let me know if I need to add anything elsewhere.

**Comment by v-prasadboke** [comment]
Hello @ekoyle, Thank you for committing the changes.
We will continue to examine this PR and provide you update by 05 September, 2023.
Thanks.

**Comment by v-prasadboke** [comment]
Hello @ekoyle, Sorry for the inconvenience but due to some Priority work the review is delayed.
I'll review this PR by 07 September, 2023 and update you about the same. Till then can you please resolve the branch conflicts.
Thanks.

**Comment by v-prasadboke** [comment]
Hello @ekoyle, [FYI] All the older parsers which were in .txt format are converted to .yaml format. And as you know you modified the .txt parser with adding all the new parsers in this solution into Corelight.txt. But the master branch still have the older parser in .txt format. 
Which is now converted to .yaml. That is what causing branch conflicts. 
So I request you to resolve the branch conflicts.

**Comment by v-prasadboke** [comment]
Hello @ekoyle, Please respond to the above comments.

**Comment by ekoyle** [comment]
@v-prasadboke I have about 100 other parsers that are still in the .txt format.  I think the simplest fix would be to get rid of the `Corelight.yaml` file for now.  Does that sound reasonable?

**Comment by v-prasadboke** [comment]
Hello @ekoyle, Sorry to say but as per latest guidelines all the parsers which are in .txt format needs to be converted to .yaml format.
You can refer to the parser of this solution which is in master branch for more clarification.

**Comment by ekoyle** [comment]
Is there any documentation on this new requirement?  How are the IDs generated?

**Comment by v-prasadboke** [comment]
Sure @ekoyle, Just give me some time I'll provide you instruction set.

**Comment by v-prasadboke** [comment]
Hello @ekoyle, We are still into this. I'll provide you the content required for .txt to .yaml conversion until then you can have a look at the .yaml parser located in this solution folder on master branch.

**Comment by v-prasadboke** [comment]
Hello @ekoyle, Have you gone through the .yaml parser of this solution which is on the master branch.

**Comment by v-prasadboke** [comment]
Hello @ekoyle, Please respond to the above comments.

**Comment by ekoyle** [comment]
It looks like someone used a script to convert all of the existing .txt parsers.  Do you know where that script would be?

Should `id` be a random UUID for each parser, or do I need to get that value from somewhere else?

**Comment by v-prasadboke** [comment]
Hello @ekoyle, We are working on the script. Will share it with you till 27 September, 2023.

**Comment by v-prasadboke** [comment]
Hello @ekoyle, Yes changes you made are good. Please resolve the branch conflicts.

**Comment by ekoyle** [comment]
I thought I resolved the conflicts with my last push. 
It looks like github is complaining about some fields that are named "password" in our logs.

**Comment by v-prasadboke** [comment]
Its because of a hardcoded value check. You can replace password with any other keyword.
Thanks

**Comment by ekoyle** [comment]
I would rather not change our schema to satisfy an arbitrary CI check.  Is this really necessary?

**Comment by v-prasadboke** [comment]
Hello @ekoyle, You can replace password with pass_s.

**Comment by ekoyle** [comment]
@v-prasadboke Other than the password ci check, what still needs completed to get this PR merged?

All of our other solutions use "password" for these fields without issue. I would like to keep this consistent if at all possible.

**Comment by v-prasadboke** [comment]
Hello @ekoyle Sorry for the inconvenience. I've dismissed the alert for Code QL. We have now repackaged the solution using V3 tool. Will update you about the same before 10 October, 2023.
Thanks.

**Comment by v-prasadboke** [comment]
Hello @ekoyle, I have changed the schema name that would remove the KQL for custom table name. 
But still a KQL check will fail for missing columns. Please add the required columns in the tables. 
And if possible can we connect tomorrow at 2:00 PM IST for Data Connector validation.
Thanks.

**Comment by v-prasadboke** [comment]
Hello @ekoyle for KQL validations I've resolved almost validation. Now It's failing for missing columns. Please add the missing columns to the custom table.

And for Data connector validation. The solution have 2 connector which have the id. Can you please look into it.

**Comment by ekoyle** [comment]
@v-prasadboke I now have all of the tables properly defined in `.script/tests/KqlvalidationsTests/CustomTables`, however the validations are failing because the hunting queries/analytic rules are using our custom parsers.

What is the proper way to define our custom parsers for these validations? For example, `corelight_conn` is a parser that redefines columns in `Corelight_v2_conn` to our standard data format, and many of our queries reference that.  For example, `Solutions/Corelight/Hunting Queries/CorelightExternalServices.yaml` has a query like this:

```
corelight_conn
  | where TimeGenerated > ago(24h)
  | where ipv4_is_private(id_orig_h) == false
  | where isnotempty(id_resp_p)
  | where history startswith 'Sh'
```

How do I tell the validations what `corelight_conn` is?

**Comment by v-prasadboke** [comment]
> @v-prasadboke I now have all of the tables properly defined in `.script/tests/KqlvalidationsTests/CustomTables`, however the validations are failing because the hunting queries/analytic rules are using our custom parsers.
> 
> What is the proper way to define our custom parsers for these validations? For example, `corelight_conn` is a parser that redefines columns in `Corelight_v2_conn` to our standard data format, and many of our queries reference that. For example, `Solutions/Corelight/Hunting Queries/CorelightExternalServices.yaml` has a query like this:
> 
> ```
> corelight_conn
>   | where TimeGenerated > ago(24h)
>   | where ipv4_is_private(id_orig_h) == false
>   | where isnotempty(id_resp_p)
>   | where history startswith 'Sh'
> ```
> 
> How do I tell the validations what `corelight_conn` is?

Validation wont fail for corelight_conn which is used in query and its basically a parser. Validation might have been failing for table defined in datatype field

**Comment by v-prasadboke** [comment]
@ekoyle you have removed custom table names Corelight.json used in data connector which is causing kql validation check
image

Can you please add a custom table named Corelight.json.
Thanks.

**Comment by v-prasadboke** [comment]
Hello @ekoyle, Thank you for committing the requested changes. Continuing with the repackaging. Will update you about the same before 19 October, 2023.

**Review by v-prasadboke — APPROVED** [review]

**Comment by v-prasadboke** [comment]
Other than that please take a look at the committed changes and if changes needed please let me know.

**Comment by ekoyle** [comment]
I have updated the release notes based on the git log.  I also removed the old data connector, as that method will no longer work with the new table structure.

**Review by v-prasadboke — APPROVED** [review]

**Review by v-prasadboke — APPROVED** [review]

**Review by v-atulyadav — APPROVED** [review]

**Pull request merged**